### PR TITLE
Users crud

### PR DESCRIPTION
**Pull request by LazarShockX**
Added a new route userInfoRoutes.
Added new delete endpoint to userRoutes.
Modified existing endpoints in applicationRoutes.
Updated middleware requireLogin and adminOnly.

### CODE REVIEW

File: __tests__/routes/applicationRoutes.test.js
Comment (madsWesth) [inline __tests__/routes/applicationRoutes.test.js:114]: Its fine to comment these out since they are broken and not because of you, just remember to notify whoever made them and tell them to fix it. On top of that you can also tell them that tests shouldnt query the database, instead they should mock the data needed.

Comment (LazarShockX) [reply __tests__/routes/applicationRoutes.test.js:114]: Det var det vi kommenterede ud sammen i tirsdag. Jeg mener, det var Lasse, der havde skrevet testene.

File: routes/courseRoutes.js
Comment (madsWesth) [inline routes/courseRoutes.js:363]: these **course** changes seem unrelated to this PR?

File: routes/userRoutes.js
Comment (madsWesth) [inline routes/userRoutes.js:22]: This commented code checks if the request is made by the owning user or an admin. If this is commented out any user can delete any other user.
This needs to be fixed!

**Review by madsWesth — CHANGES_REQUESTED** [review]

File: routes/courseRoutes.js
Comment (LazarShockX) [reply routes/courseRoutes.js:363]: Weird, we never touched the code

**Comment by LazarShockX** [comment]
Screenshot 2024-10-31 at 15 13 11

A new test created by another group that does not reflect our modifications is the reason the test fails.

**Comment by LazarShockX** [comment]
It's weird because when I fix the test in adminOnly, it fails a new test in userRoutes delete

File: __tests__/routes/applicationRoutes.test.js
Comment (Laskiri) [reply __tests__/routes/applicationRoutes.test.js:114]: Ikke mig

Comment (madsWesth) [reply __tests__/routes/applicationRoutes.test.js:114]: From this commit: https://github.com/Educado-App/educado-backend/commit/ffaa651f45aa8d4fe7dc3699218c2c5df3256fe7#diff-5f576f2166487c5d54c2e7f9d7e1b26cee7657950b54d2799024fa10fbf5927c

**Comment by madsWesth** [comment]
> It's weird because when I fix the test in adminOnly, it fails a new test in userRoutes delete. It's probably because I uncommented this in userRoutes, which for some reason was commented out before.
> 
> ```
> if (req?.tokenClaims?.id !== id && req?.tokenClaims?.role !== 'admin') {
>    return res.status(401).send({ error: errorCodes['E0002'] });
> }
> ```

users delete is dependent requireLogin/adminOnly which has been changed so now it fails. Im updating the test now so they are correct

File: __tests__/routes/applicationRoutes.test.js
Comment (Laskiri) [reply __tests__/routes/applicationRoutes.test.js:114]: Yeah I remember them now, I simply modified them a little by mocking the sendmail function since the OG tests was taking too long to complete, it was gruppe 3 that made those tests for their PBI if I recall correctly, so you should probaly tell them to fix them if they are broken.

**Comment by Laskiri** [comment]
Briefly looking through it on my phone the code looks fine, the course thing does seem a little weird, perhaps verify the file is identical with the one on dev. 
 I can try and run it later but when all testing related issues are handled i think it can go into dev with the frontend pr

**Comment by madsWesth** [comment]
> I think we should just copy the courseRoutes from dev into our branch before merging.

synchronizing this branch with dev will fix the issue. It can be done by merging dev into this branch or rebasing this branch on dev

**Comment by LazarShockX** [comment]
> > I think we should just copy the courseRoutes from dev into our branch before merging.
> 
> synchronizing this branch with dev will fix the issue. It can be done by merging dev into this branch or rebasing this branch on dev

I tried "git merge dev" into our branch and it said already up to date

**Comment by madsWesth** [comment]
> > > I think we should just copy the courseRoutes from dev into our branch before merging.
> > 
> > 
> > synchronizing this branch with dev will fix the issue. It can be done by merging dev into this branch or rebasing this branch on dev
> 
> I tried "git merge dev" into our branch and it said already up to date

checkout the dev branch and pull before going back to your own branch and merging

**Comment by LazarShockX** [comment]
> > > > I think we should just copy the courseRoutes from dev into our branch before merging.
> > > 
> > > 
> > > synchronizing this branch with dev will fix the issue. It can be done by merging dev into this branch or rebasing this branch on dev
> > 
> > 
> > I tried "git merge dev" into our branch and it said already up to date
> 
> checkout the dev branch and pull before going back to your own branch and merging

Screenshot 2024-10-31 at 18 10 39

**Comment by madsWesth** [comment]
> > > > > I think we should just copy the courseRoutes from dev into our branch before merging.
> > > > 
> > > > 
> > > > synchronizing this branch with dev will fix the issue. It can be done by merging dev into this branch or rebasing this branch on dev
> > > 
> > > 
> > > I tried "git merge dev" into our branch and it said already up to date
> > 
> > 
> > checkout the dev branch and pull before going back to your own branch and merging
> 
> Screenshot 2024-10-31 at 18 10 39

you are on the branch dev when trying to merge, so you are trying to merge dev into dev. You need to be in the branch usersCrud when you try merging the two

**Comment by LazarShockX** [comment]
> > > > > > I think we should just copy the courseRoutes from dev into our branch before merging.
> > > > > 
> > > > > 
> > > > > synchronizing this branch with dev will fix the issue. It can be done by merging dev into this branch or rebasing this branch on dev
> > > > 
> > > > 
> > > > I tried "git merge dev" into our branch and it said already up to date
> > > 
> > > 
> > > checkout the dev branch and pull before going back to your own branch and merging
> > 
> > 
> > Screenshot 2024-10-31 at 18 10 39
> 
> you are on the branch dev when trying to merge, so you are trying to merge dev into dev. You need to be in the branch usersCrud when you try merging the two
Screenshot 2024-10-31 at 18 47 30

No, I'm in usersCrud

**Comment by madsWesth** [comment]
fair enough then, i can do something that might fix it but it will make it so everybody that have the branch locally will have to delete it and refetch and i dont think thats a good idea.
Just do whatever hacky solution to fix it

**Comment by Todua992** [comment]
Try and run the diff check in git, and see what codes are different from Dev compared to yours. In that way, you can asure yourself that only the things you are modifying are different from the Dev branch. This way, you can ensure you’re not introducing unintended changes, and it keeps the codebase consistent. If everything check right, Than you are 100% sure its up to date with Dev. Another Idea, Try to make a clone of dev, and than merge your branch to Cloned-Dev branch. And than try do merge Dev to Cloned-Dev branch

**Comment by LazarShockX** [comment]
> Try and run the diff check in git, and see what codes are different from Dev compared to yours. In that way, you can asure yourself that only the things you are modifying are different from the Dev branch. This way, you can ensure you’re not introducing unintended changes, and it keeps the codebase consistent. If everything check right, Than you are 100% sure its up to date with Dev. Another Idea, Try to make a clone of dev, and than merge your branch to Cloned-Dev branch. And than try do merge Dev to Cloned-Dev branch

It's all fixed now, and I can't be bothered anymore. I like your second suggestion though and might do it next time this happens 🩵

**Comment by Todua992** [comment]
Good to hear, Thoe, sadly the tests fail. and I would like you to fix those.

1. 
image
2:
image

**Comment by LazarShockX** [comment]
No worries, I won't merge before all tests are successful 😎

**Comment by Todua992** [comment]
> No worries, I won't merge before all tests are successful 😎

raf,360x360,075,t,fafafa_ca443f4786

**Comment by LazarShockX** [comment]
Stærkt Mads 💪 Må jeg gerne merge det ind nu, så der ikke kommer flere konflikter?

**Review by Laskiri — APPROVED** [review]
Lgtm

File: __tests__/routes/applicationRoutes.test.js
Comment (LazarShockX) [reply __tests__/routes/applicationRoutes.test.js:114]: I will resolve the conversation but keep this in mind to group 3 for the next Sprint

**Pull request merged**
